tests/v2/validation/rbac/rbac_namespace_move_test.go

```diff
@@ -0,0 +1,200 @@
+	require.NoError(rb.T(), err)
+
+	// Create namespace under project read only project
+	projectReadOnlyNamespaceName := namegen.AppendRandomString("testns-")
+	projectReadOnlyNamespace, err := namespaces.CreateNamespace(rb.client, projectReadOnlyNamespaceName+"-admin-user-project-read-only-permission", "{}", map[string]string{}, map[string]string{}, rb.standardUserPR)
+	require.NoError(rb.T(), err)
+
+	// Validate Admin can move namespace from project Owner role to project read only role
+	steveAdminClient, err := rb.client.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveAdminClient = steveAdminClient
+
+	updateNS, err := getAndConvertNamespace(projectReadOnlyNamespace, rb.steveAdminClient)
+	require.NoError(rb.T(), err)
+	updateNS.Labels = projectOwnerNamespace.Labels
+	updateNS.Annotations = projectOwnerNamespace.Annotations
+
+	_, err = rb.steveAdminClient.SteveType(namespaces.NamespaceSteveType).Update(projectReadOnlyNamespace, updateNS)
+	require.NoError(rb.T(), err)
+}
+
+func (rb *NameSpaceMoveTestSuite) ValidateStandardUserCannotMoveANamespaceFromReadOnlyProject() {
+	// Create namespace under project owner project
+	projectOwnerNamespaceName := namegen.AppendRandomString("testns-")
+	projectOwnerNamespace, err := namespaces.CreateNamespace(rb.client, projectOwnerNamespaceName+"-standard-user-project-owner-permission", "{}", map[string]string{}, map[string]string{}, rb.standardUserPO)
+	require.NoError(rb.T(), err)
+
+	// Create namespace under project read only project
+	projectReadOnlyNamespaceName := namegen.AppendRandomString("testns-")
+	projectReadOnlyNamespace, err := namespaces.CreateNamespace(rb.client, projectReadOnlyNamespaceName+"-standard-user-project-read-only-permission", "{}", map[string]string{}, map[string]string{}, rb.standardUserPR)
+	require.NoError(rb.T(), err)
+
+	// Validate user cannot move namespace from project Owner role to project read only role
+	steveAdminClient, err := rb.client.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveAdminClient = steveAdminClient
+
+	updateNS, err := getAndConvertNamespace(projectReadOnlyNamespace, rb.steveAdminClient)
+	require.NoError(rb.T(), err)
+	updateNS.Labels = projectOwnerNamespace.Labels
+	updateNS.Annotations = projectOwnerNamespace.Annotations
+
+	steveStdUserclient, err := rb.standardUserClient.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveNonAdminClient = steveStdUserclient
+
+	_, err = rb.steveNonAdminClient.SteveType(namespaces.NamespaceSteveType).Update(projectReadOnlyNamespace, updateNS)
+	require.Error(rb.T(), err)
+	errStatus := strings.Split(err.Error(), ".")[1]
+	rgx := regexp.MustCompile(`\[(.*?)\]`)
+	errorMsg := rgx.FindStringSubmatch(errStatus)
+	assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+}
+
+func (rb *NameSpaceMoveTestSuite) ValidateStandardUserCannotMoveANamespaceToReadOnlyProject() {
+	// Create namespace under project owner project
```

Comment:
Have a question, since the steps 147 through 170 are same for all the three Validate functions, can we make this a single function and add validations based on the role in the project? probably a switch statement?